src/transformers/models/llava_next/modeling_llava_next.py

```diff
@@ -317,6 +363,16 @@ def __init__(self, config: LlavaNextConfig):
         self.pad_token_id = self.config.pad_token_id if self.config.pad_token_id is not None else -1
         self.post_init()
 
+    @property
+    def padding_side(self):
+        return self._padding_side
```

Comment:
self._padding_side isn't set anymore by default. So on an initial forward pass, on L533 we default to assuming we're right padding. 

I realise [this comment](https://github.com/huggingface/transformers/pull/29850#discussion_r1587694210) of mine wasn't very clear. I'd rather the whole thing was controlled via the config, which can then be overridden with setters. However, as a lot of this logic is planned to be moved out, I agree it's better to not introduce a value to the config we're going to remove. The current logic of setting I think is good. 

In the absence of taking the padding side from the config, it'd be better to have a value hard-coded at the start of the init e.g. `self._padding_side = "left"`

Reply:
Yes, now the user has to set the `padding_side` explicitly if the batch has all texts of equal length. Okay for me to bring back the hard-coded `self._padding_side = "left"` as it was in the initial state of the PR. 

If user wants to train the model, the padding_side will have to be set to "right" in model's config manually